addons/sale_purchase/models/sale_order.py

```diff
@@ -17,12 +17,25 @@ def _compute_purchase_order_count(self):
         for order in self:
             order.purchase_order_count = len(order._get_purchase_orders())
 
+    def _set_analytic_account(self):
+        purchase_orders = self._get_purchase_orders()
+        purchase_order_lines = purchase_orders.order_line.filtered(lambda line: line.sale_order_id == self)
+        for order_line in purchase_order_lines:
+            order_line.analytic_distribution = {self.analytic_account_id.id: 100}
+
     def _action_confirm(self):
         result = super(SaleOrder, self)._action_confirm()
         for order in self:
             order.order_line.sudo()._purchase_service_generation()
+            if order.analytic_account_id:
+                self._set_analytic_account()
         return result
 
+    def _create_analytic_account(self, prefix=None):
+        super()._create_analytic_account()
+        # set analytic_account fron SO to purchase order line
+        self._set_analytic_account()
+
```

Comment:
Same with this, since the set_analytic_account is called in the action confirm, I dont think it needs to be called in the ``_create_analytic_account`` method as well.

Reply:
Hello @pkyriakou 
When the sale_project module is installed, its action_confirm function is triggered first. Subsequently, the action_confirm function of the sale_purchase module is invoked through the inheritance mechanism. This sequential execution results in an issue where the analytic account, which is initially created in the sale_project module, is not set correctly. Consequently, it becomes necessary to call the set_analytic_account method twice to ensure the proper configuration of the analytic account in both manual and automatic scenarios.

Thank You:)

Reply:
It still seems strange to me :thinking: but we can either way move ahead with it